stdlib/public/core/Builtin.swift

```diff
@@ -323,12 +323,25 @@ internal func _branchHint(_ actual: Bool, expected: Bool) -> Bool {
 @usableFromInline
 @_silgen_name("_swift_objcClassUsesNativeSwiftReferenceCounting")
 internal func _usesNativeSwiftReferenceCounting(_ theClass: AnyClass) -> Bool
+
+/// Returns the class of a non-tagged-pointer Objective-C object
+@_effects(readonly)
+@_silgen_name("swift_stdlib_getObjCClassOfAllocated")
+internal func _stdlib_classOfObjCHeapObject(_ object: AnyObject) -> AnyClass
 #else
 @inlinable
 @inline(__always)
 internal func _usesNativeSwiftReferenceCounting(_ theClass: AnyClass) -> Bool {
   return true
 }
+
+@_effects(readonly)
+@_silgen_name("swift_stdlib_getObjCClassOfAllocated")
+internal func _stdlib_classOfObjCHeapObject(_ obj: AnyObject) -> AnyClass {
+  fatalError(
+    "Asking for the ObjC class of an object on non-ObjC runtimes is unsupported"
+  )
+}
```

Comment:
Is this necessary? If you omit it, then we get a compile-time rather than run-time error.

Reply:
Oh, right, good point